src/supervisor-api.ts

```diff
@@ -0,0 +1,198 @@
+import * as express from 'express';
+import { Server } from 'http';
+import * as _ from 'lodash';
+import * as morgan from 'morgan';
+
+import Config from './config';
+import { EventTracker } from './event-tracker';
+import blink = require('./lib/blink');
+import * as iptables from './lib/iptables';
+import { checkTruthy } from './lib/validation';
+
+function getKeyFromReq(req: express.Request): string | null {
+	let headerKey: string | null = null;
+	const queryKey = req.query.apikey;
+	if (queryKey != null) {
+		return queryKey;
+	}
+	const maybeHeaderKey = req.get('Authorization');
+	if (!maybeHeaderKey) {
+		return null;
+	}
+
+	const match = maybeHeaderKey.match(/^ApiKey (\w+)$/);
+	headerKey = match != null ? match[1] : null;
```

Comment:
You can change this to
```suggestion
	return match != null ? match[1] : null;
```
and avoid the need for `headerKey` altogether